src-built-in/components/UIAPI/effects/linker.ts

```diff
@@ -0,0 +1,158 @@
+import store from '../store';
+import { Channel, Channels, NameToId, Linker, linkChannelReturnObject } from '../types';
+import { updateActiveChannels, updateActives } from '../actions/linkerActions';
+
+/**
+ * Link/unlink a channel for the current component.
+ * @param channelName The channel name to toggle
+ * @param isActive Whether the channel is active or not before the toggle
+ * @param windowIdentifier The window identifier for the current component the linker window is attached to
+ */
+export const linkChannel = (channelName: string, isActive: boolean, windowIdentifier: object, actives: number) => {
+    // Limit the number of maximum active channels to 6
+    if (actives > 5 && !isActive) {
+        return Promise.reject("The number of active channels shall not exceed 6.");
+    }
+    else {
+        return new Promise((res, rej) => {
+            const callback = (err: any, data: linkChannelReturnObject) => {
+                if (err) {
+                    FSBL.Clients.Logger.system.error(`Failed to togger channel ${channelName}: ${err}`);
+                    return rej(err);
+                }
+                FSBL.Clients.Logger.system.log(`Toggle channel success. Channel name: ${channelName}`);
+                const numActiveChannels = data.channels.length;
+                store.dispatch(updateActives(numActiveChannels));
+                res(data);
```

Comment:
I find more readable to stick with `resolve` and `reject` . If you send a link to this line,  you wouldn't be sure the function is about.

Reply:
Okay I changed them to using "resolve" and "reject"